ModelFarm/src/main/java/io/deephaven/modelfarm/ModelFarmBase.java

```diff
@@ -360,6 +362,8 @@ public final boolean awaitTermination(final long timeout, final TimeUnit unit) {
                 } else {
                     allThreadsTerminated = true;
                 }
+
+                currentTime = System.currentTimeMillis();
```

Comment:
Approving, but if you wanted this code to be as tight as you can make it I would either move to `System.currentTimeMillis()` to the where (that is, `while (!allThreadsTerminated && (currentTime = System.currenttimeMillis()) < timeoutMillis) {`) or add a `break` after the lines where `allThreadsTerminate = true;` is set.